### ISSUE TITLE
[Feature Request] Follow local timelines from other instances

### ISSUE THREAD

**Issue by Hund** [tuskyapp/Tusky#874]
It would be nice if you could somehow follow the local timeline of other instances. 

I'm not sure how to best incorporate it in the client though. Perhaps via a new tab like how you view your local timeline now?

**Comment by charlag** [comment]
Hi!
I'm afraid there's no way to do that?

**Comment by Hund** [comment]
Why not? Mastolab supports it. :)

**Comment by charlag** [comment]
Oh, I didn't know that.
Yet I'm not sure that's within our scope

**Comment by charlag** [comment]
@Hund how do you think we could integrate that? I don't see any way so far. It also seems like a quite specific use case

**Comment by connyduck** [comment]
I will implement tab customization for #832, and then we can add a custom tab that is a timeline of another instance

**Comment by Hund** [comment]
That sounds like a great idea! I'm looking forward to it. :)

**Comment by cdermont** [comment]
Hi!

Adding to the request: with all the small new servers popping up now, I would like to request a higher priority for this issue if there is a technical solution. It would be great to be able to combine the "local" timeline from multiple servers to be able to follow different "bubbles". 

If somehow possible: the "local" tab could be a selection of "locals", whereby a user can add multiple instances which then form a mixed timeline, with the ability to filter on the spot. Alternatively, multiple tabs for multiple locals is a solution, but could lead to quite a few tabs if a user wants to follow multiple instances. 

An option might be not to think from the perspective of "adding new locals", but also "filtering the fediverse feed". 

Thanks!

**Comment by connyduck** [comment]
I have been thinking about this, and while fetching a foreign timeline per se should not be a problem, it will have posts, users and mentions with completely different ids than local stuff. Resolving them all is probably a very bad user experience performance wise.

I think fedilab has the feature? Can someone research how they solve this?

**Comment by fooeynet** [comment]
Yes FediLab does this, and it's a little clunky when you like or reply to a post on a different server, but it does work! Would love to see this in Tusky.

**Comment by fooeynet** [comment]
In case anyone is looking for how/where Fedilab does this:
https://github.com/caraesten/Fedilab/blob/544a89315b6f768d34775e2752406e7fae15cfc1/app/src/main/java/app/fedilab/android/helper/CrossActions.java

**Comment by beholdnec** [comment]
I would like to poke this issue. Now that the birdsite is shedding users, is there renewed interest in implementing such a feature? I'm imagining keeping one feed for tech news and another feed for cute pet pictures, assuming instances devoted to such things existed.

**Comment by TDuffinNTU** [comment]
Bumping this! I'd love to contribute myself but Kotlin is a bit beyond my experience sadly :^(
